standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/StackThreadLocal.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.hive.metastore.txn;
+
+import java.util.Stack;
+
+public class StackThreadLocal<T> {
```

Comment:
it looks like util class, why it's under txn package?

Reply:
Where should I move it? There are several util classes across the project, not sure where it belongs.

Reply:
maybe org.apache.hadoop.hive.metastore.utils?

Reply:
if we can use `counter` instead in SqlRetryHandler, this could be embedded in MultiDataSourceJdbcResource